wp-includes/rest-api/auth/class-wp-rest-key-pair.php

```diff
@@ -0,0 +1,798 @@
+<?php
+/**
+ * REST API: WP_REST_Key_Pair class.
+ *
+ * This class is responsible for adding/revoking a users API key:secret pairs.
+ *
+ * This class only allows REST authentication to the `{api_prefix}/wp/v2/token`
+ * endpoint with an API key:secret. This allows a user to be identified via the
+ * REST API without using their login credentials to generate a JSON Web Token.
```

Comment:
I see what's intended here, but as discussed in Slack I think this flow is confusing and that we'll need a request flow example (Either in js, php, whichever, saying, POST to /wp-json/tokens/generate, then take X from that response, etc…)  before we make any progress on that.